### PR TITLE
[BaseTopology] SparseGrid and SparseGridRamification cleanup

### PR DESCRIPTION
**Pull request by epernod**
Some work on cleaning the components SparseGridTopology and SparseGridRamicficationTopology: 

- Remove commented code
- Replace sout/serr by msg macro
- Remove duplicated Data in SparseGridRamicficationTopology which inherite from SparseGridTopology 
- Fix warnings (size_t, unsued variables, ...)
- Fix SparseGridTopology.scn and SparseGridRamicficationTopology.scn behavior
- Add regression_test references for both scenes and add them to the list of scene to be tested.

This PR replace an old one #706 instead of merging master.

This PR is only a aesthetic clean before a more complex PR to clean and factorize methods inside the components as well as removing the fileMesh loaded inside the component and use topology Data buffers

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-regression-tests]

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaBaseTopology/SparseGridTopology.cpp [hunk 1]
```diff
@@ -97,67 +84,36 @@ const int SparseGridTopology::cornerIndicesFromFineToCoarse[8][8]=
...
+    , facets(initData(&facets, "facets", "Input mesh facets"))
 {
     isVirtual = _isVirtual;
     _alreadyInit = false;
     _finerSparseGrid = NULL;
     _coarserSparseGrid = NULL;
     _usingMC = false;
 
+    addAlias(&seqPoints, "vertices");
+
```
Comment (guparan) [inline SofaKernel/modules/SofaBaseTopology/SparseGridTopology.cpp]: Do we really need this alias?

File: SofaKernel/modules/SofaBaseTopology/SparseGridTopology.cpp [hunk 2]
```diff
@@ -97,67 +84,36 @@ const int SparseGridTopology::cornerIndicesFromFineToCoarse[8][8]=
...
-    _coarserSparseGrid = NULL;
-    _usingMC = false;
-
-    _regularGrid = sofa::core::objectmodel::New<RegularGridTopology>();
-
     //Add alias to use MeshLoader
-    addAlias(&vertices,"position");
-    addAlias(&input_triangles,"triangles");
-    addAlias(&input_quads,"quads");
+    addAlias(&seqPoints, "vertices");
```
Comment (guparan) [inline SofaKernel/modules/SofaBaseTopology/SparseGridTopology.cpp]: Do we really need this alias?

**Pull request merged**